### ISSUE TITLE
Investigate how to make Companion connection persistent

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#1192]
**What to investigate**
For some reason, the companion connection is automatically closed after some inactivity. I'm now sure why that is and I haven't looked that close into it, especially not after adding support for sessions (which might have changed something). This needs to be investigated as a connection must be maintained in order to continuously receive push updates, once that has been implemented (#1191).

**Expected outcome**
What needs to be done to to maintain a connection (+ updated docs).

**Comment by postlund** [comment]
I believe this has something to do with me not starting sessions properly. There are errors in the log regarding "session start timeout", so I should probably look into that.

**Comment by postlund** [comment]
So, I've made some progress. One of the reason things are failing is because the connection never gets a "peer id", which I assume is an important identifier. By sending a `_systemInfo` containing `_i` (in content) I can make the connection get a peer id and after doing so, the connection seems to stay open. I've kept a connection open for about fifteen minutes now and that was not even close to possible before.

I've also managed to subscribe to some kind of updates with this:

```yaml
{'_i': '_interest', '_x': 1234, '_t': '1', '_c': {'_regEvents': ['_iMC']}}
```

My belief is that this corresponds to media controls. I can see things like this in the log:

    förval	22:52:30.101516+0200	rapportd	MediaControl flags changed: < Play Pause NextTrack PrevTrack FFwd Rwd Repeat Shuffle >
    förval	22:52:30.101699+0200	rapportd	11730: SendEventID '_iMC', XID 0x82861491, <6e1066e28ae7f388830c981d656d1dc5> (53 bytes)

When streaming something from the Music app via AirPlay and press next/prev, I get updates like this:

    {'_i': '_iMC', '_x': 2189825169, '_c': {'_mcF': 255}, '_t': 1}

I however don't handle events properly in my Companion implementation, so I can't deal with further updates. Need to fix that. But based on what I've seen, my bet is that AirPlay is where things are happening. I can probably do some things via Companion, but I need to put some effort into AirPlay 2 and see what is happening at that party.
